index.test-d.ts

```diff
@@ -28,3 +28,14 @@ expectType<{[key in 'foo-bar']: true}>(
 expectType<{[key in 'foo-bar']: true}>(
 	camelcaseKeys({'foo-bar': true}, {stopPaths: ['foo']}),
 );
+
+interface SomeObject {
+	someProperty: string;
+}
+
+const someObj: SomeObject = {
+	someProperty: "hello",
+};
+
+expectType<SomeObject>(camelcaseKeys(someObj));
+expectType<SomeObject[]>(camelcaseKeys([someObj]));
```

Comment:
@haakondr I would also throw in an additional tests doing the same thing above, only using a type alias (`type = ...`) instead of an interface.

Reply:
Added below. Is this what you meant?

 (amended commit and force pushed on the same commit)

Reply:
Yeah, that's what I had in mind.
@sindresorhus ^^